lib/utils/getRetirement/index.ts

```diff
@@ -25,33 +25,54 @@ export const createRetirementStorageContract = (
   );
 };
 
-export const getRetirements = async (
-  beneficiaryAdress: string
-): Promise<RetirementsResult> => {
+export const getRetirementTotalsAndBalances = async (params: {
+  address: string;
+}): Promise<RetirementsTotalsAndBalances> => {
   try {
     const provider = getJsonRpcProvider();
-    const storageContract = createRetirementStorageContract(provider);
+    const retirementStorageContract = createRetirementStorageContract(provider);
 
     const [
       totalRetirements,
-      totalCarbonRetired,
-      totalClaimed,
-    ]: RetirementTotals = await storageContract.getRetirementTotals(
-      beneficiaryAdress
+      totalTonnesRetired,
+      totalTonnesClaimedForNFTS,
+    ]: RetirementTotals = await retirementStorageContract.getRetirementTotals(
+      params.address
+    );
+    const bct = await retirementStorageContract.getRetirementPoolInfo(
+      params.address,
+      addresses["mainnet"].bct
+    );
+    const mco2 = await retirementStorageContract.getRetirementPoolInfo(
+      params.address,
+      addresses["mainnet"].mco2
+    );
+    const nct = await retirementStorageContract.getRetirementPoolInfo(
+      params.address,
+      addresses["mainnet"].nct
+    );
+    const ubo = await retirementStorageContract.getRetirementPoolInfo(
+      params.address,
+      addresses["mainnet"].ubo
+    );
+    const nbo = await retirementStorageContract.getRetirementPoolInfo(
+      params.address,
+      addresses["mainnet"].nbo
```

Comment:
Please refactor to promise.all and then it's good to merge!

Reply:
done